components/kyma-environment-broker/internal/environmentscleanup/service.go

```diff
@@ -59,25 +61,28 @@ func (s *Service) PerformCleanup() error {
 		runtimeID, ok := shoot.Annotations[shootAnnotationRuntimeId]
 		if !ok {
 			err = errors.New(fmt.Sprintf("shoot %q has no runtime-id annotation", shoot.Name))
-			log.Error(err)
-			result = multierror.Append(result, err)
+			s.logger.Error(err)
 			continue
 		}
 		runtimeIDsToDelete = append(runtimeIDsToDelete, runtimeID)
 	}
-	log.Infof("Runtime IDs to process: %v", runtimeIDsToDelete)
+	s.logger.Infof("Runtime IDs to process: %v", runtimeIDsToDelete)
+
+	if len(runtimeIDsToDelete) == 0 {
+		return result.ErrorOrNil()
+	}
 
 	instancesToDelete, err := s.getInstancesForRuntimes(runtimeIDsToDelete)
 	if err != nil {
-		log.Error(errors.Wrap(err, "while getting instance IDs for Runtimes"))
+		s.logger.Error(errors.Wrap(err, "while getting instance IDs for Runtimes"))
 		result = multierror.Append(result, err)
 	}
 
 	for _, instance := range instancesToDelete {
-		log.Infof("Triggering environment deprovisioning for instance ID %q", instance.InstanceID)
+		s.logger.Infof("Triggering environment deprovisioning for instance ID %q", instance.InstanceID)
 		currentErr := s.triggerEnvironmentDeprovisioning(instance)
 		if currentErr != nil {
-			log.Error(errors.Wrapf(currentErr, "while triggering deprovisioning for instance ID %q", instance.InstanceID))
+			s.logger.Error(errors.Wrapf(currentErr, "while triggering deprovisioning for instance ID %q", instance.InstanceID))
 			result = multierror.Append(result, currentErr)
 		}
 	}
```

Comment:
as discussed, we could extract formatting function:
```
 func(i []error) string {
			var s []string
			for _, v := range i {
				s = append(s, v.Error())
			}
			return strings.Join(s, ", ")
		}
```
and assign it to `result.ErrorFormat` each time it is going to be returned